server/src/main/java/org/opensearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -149,6 +168,202 @@ public TransportShardBulkAction(
         );
         this.updateHelper = updateHelper;
         this.mappingUpdatedAction = mappingUpdatedAction;
+        this.transportPrimaryTermValidationAction = ACTION_NAME + "[ptv]";
+
+        transportService.registerRequestHandler(
+            transportPrimaryTermValidationAction,
+            executor,
+            true,
+            true,
+            PrimaryTermValidationRequest::new,
+            this::handlePrimaryTermValidationRequest
+        );
+    }
+
+    protected void handlePrimaryTermValidationRequest(
+        final PrimaryTermValidationRequest request,
+        final TransportChannel channel,
+        final Task task
+    ) {
+        ActionListener<ReplicaResponse> listener = new ChannelActionListener<>(channel, transportPrimaryTermValidationAction, request);
+        final ShardId shardId = request.getShardId();
+        assert shardId != null : "request shardId must be set";
+        IndexShard replica = getIndexShard(shardId);
+        try {
+            new PrimaryTermValidationReplicaAction(listener, replica, (ReplicationTask) task, request).run();
+        } catch (RuntimeException e) {
+            listener.onFailure(e);
+        }
```

Comment:
Curious what happens on non `RunTimeException`

Reply:
If it is non `RunTimeException`, then it would either be a checked Exception or Error. Checked exceptions needs to be handled explicitly and error anyways should not be caught anyways. That should take care of all possible Throwable implementations.